requests-for-comments/2023-10-27-base3000-launch.md

```diff
@@ -0,0 +1,34 @@
+# Base 3000 & Notebooks launch
+
+We want to launch the ‘first version’ of PostHog 3000 after the next sprint, which means the week starting November 13th (+ 1 week if needed). We have decided to **include Notebooks** in this launch, so that we can have a big launch that combines functionality and design changes.
```

Comment:
Can we get some detail on what the launch will functionally look like to users? 

I'm not clear if PostHog 3000 will be a opt-in beta for them, something they can toggle on/off, controlled with early access management, or if we're just going to switch users to a new UI without warning? 

I'd actually really appreciate a demo of how this will work ahead of launch, and I suspect @andyvan-ph @Lior539 and @ivanagas may want eyes on it too.

Reply:
1) PostHog 3000 should be rolled out to everyone, not as a feature preview, in my opinion. Otherwise it will be hard to time launch announcement and actual launch
2) Having something prominent in the UI that announces the change, maybe with a link to the blog post is something we could do
3) Giving people the option to revert to the old design -> Maybe, but this first PostHog 3000 version is free from big controversial changes, so I think we should just change it and people will get used to it
4) A demo of the 'launch version', highlighting what has changed is a good idea. @benjackwhite @Twixes is this something one of you could do in one of the next All Hands?

Reply:
10000% for making this a feature preview, no question - at least for a sprint or two (but ideally much longer). While we've made a bunch of structural changes, this rollout doesn't demonstrate enough value to justify changing the entire UI. We have a ton of other changes that won't be rolled out yet:

- New query builder
- Deep dives (whether it's a sidebar or tabs, the ability to look into data without being committed and having to use the back button)
- Panes
- Tabs

On top of that, the left pane is entirely unusable (my fault). We have a lot of massaging to do with it to really make it feel more useful than the current index views. As of now, I personally feel lost and like it's harder to use. So until we at least get to a point where ** _I_ ** feel comfortable, I do not think we should be forcing this update on anyone.

This is another example of something we're prone to shipping [entirely too early](https://github.com/PostHog/company-internal/issues/1183) and it can only damage us forcing it on everyone early vs taking a little more time for refinement.

I understand that it's hard to maintain the current UI and 3000 forever, but it's definitely worth putting off a full launch until we can solve some of the basic UX issues and also put together a good story about what our long-term vision is, so that people can buy into it.

Reply:
Just chiming in to add a strong +1 Cory's POV. Purely based on what I can see right now, PostHog 3000 doesn't feel ready for release outside of Feature Preview. 

I'd like to:

1. Have more time to dogfood it and provide feedback internally (I wasn't aware it had substantively changed until recently and haven't used it much personally).

2. Actively recruit some friendly users to get their take on it.

This feels way bigger than launching Surveys, for example, and the potential for a backlash is **huge**.

Reply:
(Please don't base this on how the current flag feels like. We are actively working this sprint on getting the things we think are releasable into a releasable, stable state 😅 - all the more subjective things are purposefully being moved to follow up releases, after much more intense iteration and internal testing).

Reply:
Cool, that's fine, but I'd still like to tread with a tiny bit more caution + have more time to use it in its "ready" state. The UI is something literally every user interacts with, so it requires a bit more belt and braces than a typical beta launch.

Reply:
We are likely going to stretch the timeline a bit @andyvan-ph to have enough time for your points: Familiarise the team with the changes, look out for things that don't work, and potentially run some usability tests with external people.

I'll update this RFC on Friday, once we had the chance to meet as a team and agree on a plan forward.

Reply:
Sounds good, thanks. 😊